group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/TopicMetadata.java

```diff
@@ -40,23 +44,31 @@ public class TopicMetadata {
      */
     private final int numPartitions;
 
+    /**
+     * Map of every partition to a set of its rackIds.
+     * If the rack information is unavailable, pass an empty set.
+     */
+    private final Map<Integer, Set<String>> partitionRackInfo;
```

Comment:
I am wondering if we should introduce `PartitionMetadata` instead of using `Set<String>` directly. I am not sure if it is worth it though.

Reply:
I also thought about it, but realized its not that necessary since PartitionMetadata is used to finally pass info to the assignor from targetAssignmentBuilder but TopicMetadata is used to get info from records till targetAssignmentBuilder.